### PR TITLE
gh-98776: Fix make regen-test-levenshtein for out-of-tree builds

### LINKED ISSUE #98776: Out of tree build of Python 3.12.0a1+ breaks regen-test-levenshtein
# Bug report

In Fedora, when building Python 3.12.0a1 with Python 3.12.0a1 for regen, we found that regen-test-levenshtein fails. It is not necessary to use Python 3.12.0a1 for regen to trigger the failure:

To reproduce:

```
# from the root cpython directory
$ mkdir outoftree
$ cd outoftree
$ ../configure
$ make regen-test-levenshtein
...
# Regenerate Lib/test/levenshtein_examples.json
python3.10 ../Tools/build/generate_levenshtein_examples.py Lib/test/levenshtein_examples.json
Traceback (most recent call last):
  File ".../cpython/outoftree/../Tools/build/generate_levenshtein_examples.py", line 70, in <module>
    main()
  File ".../cpython/outoftree/../Tools/build/generate_levenshtein_examples.py", line 65, in main
    with open(output_path, "w") as f:
FileNotFoundError: [Errno 2] No such file or directory: '.../cpython/outoftree/Lib/test/levenshtein_examples.json'
make: *** [Makefile:1092: regen-test-levenshtein] Error 1
```

# Your environment

- CPython versions tested on: Python 3.12.0a1 or the current main branch
- Operating system and architecture: Fedora Linux x86_64

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/python/cpython/issues/98776

* Issue: gh-98776

Automerge-Triggered-By: GH:erlend-aasland

### CODE REVIEW

File: Misc/NEWS.d/next/Build/2022-10-27-19-47-31.gh-issue-98776.lt_UOG.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+When building Pyton out-of-tree, don't crash during ``make
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Build/2022-10-27-19-47-31.gh-issue-98776.lt_UOG.rst]:
```suggestion
When building Python out-of-tree, don't crash during ``make
```

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Should this be backported?

**Comment by hroncok** [comment]
IIRC this only exists in 3.12.

File: Misc/NEWS.d/next/Build/2022-10-27-19-47-31.gh-issue-98776.lt_UOG.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+When building Python out-of-tree, don't crash during ``make
+regen-test-levenshtein``.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Build/2022-10-27-19-47-31.gh-issue-98776.lt_UOG.rst]:
```suggestion
Fix ``make regen-test-levenshtein`` for out-of-tree builds.
```

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**